crates/driver/src/tests/setup/mod.rs

```diff
@@ -777,35 +852,85 @@ impl Setup {
         // Create the necessary components for testing.
         let blockchain = Blockchain::new(blockchain::Config {
             pools,
-            trader_address,
-            trader_secret_key,
+            main_trader_secret_key: trader_secret_key,
+            trader_secret_keys: self
+                .solvers
+                .iter()
+                .filter_map(|solver| {
+                    if solver.get_configure_for_jit_orders() {
+                        Some(*solver.private_key.clone())
+                    } else {
+                        None
+                    }
+                })
+                .collect::<Vec<_>>(),
             solvers: self.solvers.clone(),
             settlement_address: self.settlement_address,
             rpc_args: self.rpc_args,
         })
         .await;
         let mut solutions = Vec::new();
+        let mut specific_solutions: HashMap<String, Vec<blockchain::Solution>> = HashMap::new();
         for solution in self.solutions {
             let orders = solution
                 .orders
                 .iter()
-                .map(|solution_order| orders.iter().find(|o| o.name == *solution_order).unwrap());
-            solutions.push(blockchain.fulfill(orders, &solution).await);
+                .flat_map(|solution_order| orders.iter().filter(|o| o.name == *solution_order))
```

Comment:
Looks like we are assembling "complete" solutions with data from 2 different places. I would expect a solution returned from a solver should already contain the JIT order it's supposed to contain according to this logic.